### DISCUSSION TITLE
Outpost and Road Events

### DISCUSSION THREAD

**Discussion by Draazi** [Lurkars/gloomhavensecretariat#531] [Q&A]
Hi there,

Apologies if this information is already available somewhere, but I looked through the documentation and the discussions and couldn't find my answer (or I overlooked it, which is entirely possible).

We started a Frosthaven campaign yesterday (finally, yeah) and are using GHS for tracking turns and the campaign and character progress. We love it, by the way ! Great work !!!

It helped us so much that we had actually already cleared the table and put everything back in the box when we needed to do the outpost phase. Which went ok based on what we could see on the campaign/party sheet except for the Outpost event we needed to draw, so we decided to postpone that until our next playsession :-)

I've been looking at documentation and have seen that support for event cards is Planned Next, so I have 2 questions : 
1) Is there a rough ETA for this functionality ? 
2) How do other groups manage the tracking of their events and event decks in the mean time ? Just use the physical cards ? Or is there another app people are using ? 

Thanks in advance for any answers/advice !

Draazi.

**Comment by Lurkars** [comment]
There won't be any ETA. The big problem is just the data. I have to manually type every event or at least edit and copy every event (for base GH) to get just the basic data and an overview. This is a so long taking task and I already asks for community contributions here. No idea how long this will take. Can be a month, can be a year. So you shouldn't wait for it.

Personally, we use just the physical deck. The only app I know supporting events is [Gloomhaven Full Stack](https://gloomhaven.smigiel.us/v2/), but I didn't do any research on is, so possible that there are other/better solutions.

**Reply by Draazi** [reply]
I honestly have no idea of the amount of work, but if you could maybe send me a bit more details on what exactly needs to happen/be done, I could potentially help out a bit.

**Reply by Lurkars** [reply]
So you can have a look at the existing data for GH:

Here is the card data that defines the possible options and requirements:
https://github.com/Lurkars/gloomhavensecretariat/blob/e0f28e644212971d5b7df283b5aca1533a993869/data/gh/events.json#L3

And here are the corresponding card texts:
https://github.com/Lurkars/gloomhavensecretariat/blob/e0f28e644212971d5b7df283b5aca1533a993869/data/gh/label-spoiler/en.json#L622

For Gloomhaven I have this source: https://github.com/MatchbookLab/gloomhavendb/blob/master/src/data/events.ts

For all other editions, I can only refer to the Worldhaven repository, which has all the cards as images:
https://github.com/any2cards/worldhaven/tree/master/images/events

So all these cards have to be converted to the given format, also with the caveat that I only know possible rewards from GH events, so possible that I still have to implement more keywords to make it possible (but this is a minimal problem compared to *digitalising* all these events first.

**Comment by Lurkars** [comment]
Last weeks a lot of happened here. I would say for base Gloomhaven (and maybe FC) event cards are ready to use now in [v0.123.0](https://github.com/Lurkars/gloomhavensecretariat/releases/tag/v0.123.0). For Frosthaven (and other edition), a lot of work still has to be done. But I am already happy about feedback here.

**Reply by tirmey** [reply]
I think I can help with Jaws events, if it's still necessary.

**Reply by Lurkars** [reply]
Thanks, you can just check the events for errors here
https://gloomhaven-secretariat.de/tools/events?edition=jotl&type=city in general the JotL events are also done. Some outcome effect wording may not match exactly, because it differs from base GH and I didn't want to make it too complicated, so this is on purpose and no issue for me.

Otherwise, I am currently working on the FH events and are in the middle of summer road. Winter events are missing.

On CS/ToA events no work yet done, just the parsed infos.
